platform/osgi/platform-osgi-configadmin/src/main/java/org/codice/felix/cm/file/EncryptingWrappedPersistenceManager.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.felix.cm.file;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Dictionary;
+import java.util.Enumeration;
+import java.util.HashSet;
+import java.util.Hashtable;
+import java.util.Set;
+
+import org.apache.felix.cm.PersistenceManager;
+
+/**
+ * Performs encryption.
+ */
+public class EncryptingWrappedPersistenceManager extends WrappedPersistenceManager {
+
+    private final CopiedEncryptionService encryptionService;
+
+    private final Set<String> excludedProperties = propertiesExcludedFromEncryption();
+
+    public EncryptingWrappedPersistenceManager(PersistenceManager persistenceManager) {
+        super(persistenceManager);
+        this.encryptionService = new CopiedEncryptionService();
+    }
+
+    public EncryptingWrappedPersistenceManager(PersistenceManager persistenceManager,
+            CopiedEncryptionService encryptionService) {
+        super(persistenceManager);
+        this.encryptionService = encryptionService;
+    }
+
+    @Override
+    public Dictionary load(String pid) throws IOException {
+        return decryptAndReturnDictonary(super.load(pid));
+    }
+
+    @Override
+    public void store(String pid, Dictionary properties) throws IOException {
+        super.store(pid, encryptAndReturnDictonary(properties));
+    }
+
+    //The 2.3.7 version of the maven-bundle-plugin fails on lambdas
```

Comment:
We're using the 2.3.7 version of the maven-bundle-plugin since that's what Felix's ConfigAdmin 1.8.14 is using. That version of the plugin fails on Java 8 code, which is why we had to implement this method the Java 7 way. We'll make the comment more clear.

Reply:
There's a bug (array index out of bounds) in version 2.3.7 of the bundle plugin:
https://issues.apache.org/jira/browse/FELIX-4005

I tested bumping the version of the maven-bundle-plugin a few times but encountered different errors, such as the dropped support for running the plugin in static scope and blatantly ignoring my directives for skipping imports. Eventually I decided it wasn't worth the immediate effort and proceeded with Java 7 compliant code only.